src/key-value-engine/B-Tree-vs-Log-Structured-Merge-Tree.md

```diff
@@ -0,0 +1,130 @@
+
+The number of disk reads for query is at most \\(O(log_BN/B)\\) which is the depth of the tree. 
+
+### Level-Based LSM-tree
+
+In the Level-based LSM-tree, data is organized into levels. Each level contains one sorted run. Data starts in level 0, then gets merged into the level 1 run. Eventually the level 1 run is merged into the level 2 run, and so forth. Each level is constrained in its sizes. Growth factor \\(k\\) is specified as the magnification of data size at each level.
+
+$$
+level_i = level_{i-1} * k
+$$
+We can analyze the Level-based LSM-tree as follows. If the growth factor is \\(k\\) and the smallest level is a single file of size \\(B\\), then the number of levels is 
+
+$$
+Θ(log_kN/B)
+$$
+where \\(N\\) is the size of the database. In order to simplify the analysis, we assumes that database size is stable and grows slowly over time, so that the size of database will be nearly equal as the size of last level.
+
+#### Write Amplification
+
+Data must be moved out of each level once, but data from a given level is merged repeatedly with data from the previous level. On average, after being first written into a level, each data item is remerged back into the same level about \\(k/2\\) times. So the total write amplification is 
+$$
+Θ(k*log_kN/B)
+$$
+
+#### Read Amplification
+
+To perform a short range query in the cold cache case, we must perform a binary search
+on each of the levels.
+
+For the highest \\(level_i\\), the data size is \\(O(N)\\), so that it performs \\(O(logN/B)\\) disk reads. 
+
+For the previous \\(level_{i-1}\\), the data size is is \\(O(N/k)\\), so that it performs \\(O(log(N/(kB))\\) disk reads. 
+
+For \\(level_{i-2}\\), the data size is \\(O(N/k^2)\\), so that it performs \\(O(log(N/k^2B)\\) disk reads.
+
+…
+
+For \\(level_{i-n}\\), the data size is \\(O(N/k^n)\\), so that it performs \\(O(log(N/k^nB)\\) disk reads.
+
+So that the total number of disk reads is 
+
+$$
+R = O(logN/B) + O(log(N/(kB)) + O(log(N/k^2B) + ... + O(log(N/k^nB) + 1 = O((log^2N/B)/logk)
+$$
+
+## Summary
+
+The following table shows the summary of various kinds of amplification:
+
+|    Data Structure    | Write Amplification |    Read Amplification    |
+| :------------------: | :-----------------: | :----------------------: |
+|       B+ tree        |     \\(Θ(B)\\)      |    \\(O(log_BN/B)\\)     |
+| Level-Based LSM-tree | \\(Θ(klog_kN/B)\\)  | \\(Θ((log^2N/B)/logk)\\) |
+
+> Table 1. A summary of the write amplification and read amplification for range queries.
+
+Through comparing various kinds of amplification between B+ tree and Level-based LSM-tree, we can come to a conclusion that Level-based LSM-tree has a better write performance than B+ tree while its read performance is not as good as B+ tree. The main purpose for TiKV to use LSM-tree instead of B-tree as its underlying storage engine is because using cache technology to promote read performance is much easier than promote write performance.
```

Comment:
```suggestion
Through comparing various kinds of amplification between B+ tree and Level-based LSM-tree, we can come to a conclusion that Level-based LSM-tree has a better write performance than B+ tree while its read performance is not as good as B+ tree. The main purpose for TiKV to use LSM-tree instead of B-tree as its underlying storage engine is because using cache technology to promote read performance is much easier than promoting write performance.
```